src/api/utils.ts

```diff
@@ -51,3 +52,34 @@ export const stringToJSON = <T>(string: string | undefined): T | undefined => {
 };
 
 export const removeProtocolFromURL = (url = '') => url.replace(/^(http|https):\/\//, '');
+
+type FetchApiEndpointParams<T> = {
+  apiCallback: () => AxiosPromise<T>;
+  onSuccess: (data: T) => void;
+  errorTitle: string;
+  onError?: (e: AxiosError<T>) => void;
+};
+
+export const fetchApiEndpoint = <T>({
```

Comment:
I plan to refactor similar code at other places to use this function.

Reply:
It's not a must to have it. The pattern repeats, so the idea was to abstract it. I understand your point, I am not big fan of over-abstracting either. So let's postpone such refactoring to a follow-up.

Reply:
removed from here